### PR TITLE
timeout for tokens

### PR DESCRIPTION
**Pull request by Wo-m**

### CODE REVIEW

File: src/main/java/com/deco/tickexperience/service/TokenService.java [hunk 1]
```diff
@@ -1,32 +1,65 @@
...
+
+        if (activeUser.getLastActiveTime().isBefore(LocalDateTime.now().minusSeconds(tokenTimeout))) {
+            log.info("Token expired: {}", token);
+            tokens.remove(token);
+            return false;
+        }
+
+        // update token time
+        activeUser.setLastActiveTime(LocalDateTime.now());
+        return true;
```
Comment (hcmgr) [inline src/main/java/com/deco/tickexperience/service/TokenService.java:63]: ye this is nice

File: src/main/java/com/deco/tickexperience/model/ActiveUser.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import lombok.Data;
+
+import java.time.LocalDateTime;
+
+@Data
+@AllArgsConstructor
+public class ActiveUser {
+    private Long userId;
+    private LocalDateTime lastActiveTime;
+}
```
Comment (hcmgr) [inline src/main/java/com/deco/tickexperience/model/ActiveUser.java:13]: Did you add this to the db model as well?

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -17,3 +17,7 @@ spring:
 # Application Constants
 apiPrefix: /api
 apiAuthedPrefix: /api/authed
+
+token:
+    timeout: 900  # seconds, 15 minutes
+    length: 16 # bytes
```
Comment (hcmgr) [inline src/main/resources/application.yml:23]: so we're treating application.yml as our config file from now on?

**Review by hcmgr — APPROVED** [review]
gimble

File: src/main/java/com/deco/tickexperience/model/ActiveUser.java (hunk 2 above)
Comment (Wo-m) [reply src/main/java/com/deco/tickexperience/model/ActiveUser.java:13]: Don't think we need to persist ActiveUsers, this was more just for convenience and easier to read code. Could use a java Pair instead (shitty java version of tuples) but same overhead and harder to read.

File: src/main/resources/application.yml (hunk 3 above)
Comment (Wo-m) [reply src/main/resources/application.yml:23]: application.yml is config. It just also stores custom config like this, and also config for dependencies like db stuff. I believe there is a way to seperate into multiple ymls and have a parent one but never done it.

**Pull request merged**